common/app/views/support/cleaner/AmpEmbedCleaner.scala

```diff
@@ -60,6 +60,33 @@ case class AmpEmbedCleaner(article: Article) extends HtmlCleaner {
 
   }
 
+  object AmpSoundcloud {
+    def createElement(document: Document, trackId: String): Element = {
+      val soundcloud = document.createElement("amp-soundcloud")
+      soundcloud.attr("data-trackid", trackId)
+      soundcloud.attr("data-visual", "true")
+      soundcloud.attr("height", "300") // height is necessary if data-visual == true
+    }
+
+    def getTrackIdFromUrl(soundcloudUrl: String): String = {
+      val pattern = "soundcloud.com%2Ftracks%2F(\\d+)".r
+      pattern.findAllIn(soundcloudUrl).matchData.map { matches => matches.group(1) }.mkString
```

Comment:
The trouble with a simple match is that I want to capture the `(\\d+)` (which will be the track ID). I'm I missing something really obvious?